src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,287 @@
+     */
+    public boolean hasFinishedLoading() {
+        return !this.isLoading;
+    }
+
+    /**
+     * Prints a list of tasks on hand.
+     *
+     * @return Lists of tasks.
+     */
+    public String getList() {
+        return this.tasks.getList();
+    }
+
+    /**
+     * Marks the task as completed.
+     *
+     * @param i Index of the task to be marked as completed.
+     * @return Associated message from Duke.
+     */
+    public String markTask(int i) {
+        return this.tasks.markTask(i);
+    }
+
+    /**
+     * Marks the task as not completed.
+     *
+     * @param i Index of the task to be marked as not completed.
+     * @return Associated message from Duke.
+     */
+    public String unmarkTask(int i) {
+        return this.tasks.unmarkTask(i);
+    }
+
+    /**
+     * Adds a task without deadline.
+     *
+     * @param s Task description.
+     * @return Associated message from Duke.
+     */
+    public String addTodo(String s) {
+        return this.tasks.addTodo(s);
+    }
+
+    /**
+     * Adds a task with deadline.
+     *
+     * @param s Task description.
+     * @param d Deadline in LocalDate format.
+     * @return Associated message from Duke.
+     */
+    public String addDeadline(String s, LocalDate d) {
+        return this.tasks.addDeadline(s, d);
+    }
+
+    /**
+     * Adds a task with deadline.
+     *
+     * @param s Task description.
+     * @param d Deadline in String format.
+     * @return Associated message from Duke.
+     */
+    public String addDeadline(String s, String d) {
+        return this.tasks.addDeadline(s, d);
+    }
+
+    /**
+     * Adds an event.
+     *
+     * @param s Event description.
+     * @param d Event time in LocalDate format.
+     * @return Associated message from Duke.
+     */
+    public String addEvent(String s, LocalDate d) {
+        return this.tasks.addEvent(s, d);
+    }
+
+    /**
+     * Adds an event.
+     *
+     * @param s Event description.
+     * @param d Event time in String format.
+     * @return Associated message from Duke.
+     */
+    public String addEvent(String s, String d) {
+        return this.tasks.addEvent(s, d);
+    }
+
+    /**
+     * Deletes the task from the list.
+     *
+     * @param i Index of the task to be deleted.
+     * @return Associated message from Duke.
+     */
+    public String deleteTask(int i) {
+        return this.tasks.deleteTask(i);
+    }
+
+    /**
+     * Parses a message written to the Duke bot.
+     *
+     * @param s Message to be parsed.
+     * @return Associated message from Duke.
+     */
+    public String parse(String s) {
+        return this.parser.parse(this, s, this.hasFinishedLoading());
+    }
+
+    /**
+     * Prints a list of tasks with the matching input string.
+     *
+     * @param s String to request for tasks with the matching string.
+     * @return List of tasks with matching string.
+     */
+    public String find(String s) {
```

Comment:
Would it be better if the method naming was a bit more descriptive?

Reply:
Yes, I will change it to findTask, thank you